packages/plus/src/components/SteppedListCard/helper.ts

```diff
@@ -0,0 +1,52 @@
+import { createContext } from 'react'
+
+type ContextType = {
+  currentStep: number
+  setCurrentStep: React.Dispatch<React.SetStateAction<number>>
+  numberOfSteps: number
+  done: boolean[]
+  setDone: React.Dispatch<React.SetStateAction<boolean[]>>
+  setHidden: React.Dispatch<React.SetStateAction<boolean>>
+}
+
+export const Data = createContext<ContextType>({
+  currentStep: 1,
+  setCurrentStep: () => {},
+  numberOfSteps: 1,
+  done: [false, false, false, false, false],
+  setDone: () => {},
+  setHidden: () => {},
+})
+
+type NextStepProps = {
+  completed: boolean
+  setCompleted: React.Dispatch<React.SetStateAction<boolean[]>>
+  stepNumber: number
+  setCurrentStep: React.Dispatch<React.SetStateAction<number>>
+  numberOfSteps: number
+  setHidden: React.Dispatch<React.SetStateAction<boolean>>
+  done: boolean[]
+}
+
+/**
+ * Go to next step with or without validation of the current step
+ * Close the component for the last step
+ */
+export const nextStep = ({
+  completed,
+  setCompleted,
+  stepNumber,
+  setCurrentStep,
+  numberOfSteps,
+  setHidden,
+  done,
+}: NextStepProps) => {
+  const tempDone = done
+  tempDone[stepNumber - 1] = completed
```

Comment:
There is a mutation here we try to avoid doing them as there is a risk of side effects, better do the following:

```ts
const tempDone = done.map((item, index) => index === stepNumber - 1 ? completed : item);
```

If you want to check about it here is a good article: https://dev.to/amissah17/ultimate-guide-to-mutations-in-javascript-2a80#:~:text=Mutation%20means%20a%20change%20in%20the%20original%20code%20or%20data.&text=In%20the%20example%2C%20notice%20that,at%20another%20example%20involving%20strings.